tests/native_api_tests/test_api_model.cpp

```diff
@@ -513,4 +537,133 @@ TEST_CASE("API Model") {
     }
 }
 
+/*
+
+source_1 -- node_0 --load_2
+
+source and node inputs are row based.
+load input is either row based or columnar.
+load update is row based / columnar and densse / sparse.
+
+invalid_id_t tests are for testing the error handling of the model when the id is not found in the update dataset.
+optional_id_t tests are for testing the model when the id is not added to the update dataset.
+
+*/
+TEST_CASE_TEMPLATE(
+    "API update id tests", T, TypeCombo<row_t, row_t, dense_t, with_id_t>, TypeCombo<row_t, row_t, sparse_t, with_id_t>,
+    TypeCombo<columnar_t, columnar_t, dense_t, with_id_t>, TypeCombo<columnar_t, columnar_t, sparse_t, with_id_t>,
+    TypeCombo<columnar_t, row_t, dense_t, with_id_t>, TypeCombo<columnar_t, row_t, sparse_t, with_id_t>,
+    TypeCombo<row_t, columnar_t, dense_t, with_id_t>, TypeCombo<row_t, columnar_t, sparse_t, with_id_t>,
+    TypeCombo<row_t, row_t, dense_t, invalid_id_t>, TypeCombo<row_t, row_t, sparse_t, invalid_id_t>,
+    TypeCombo<columnar_t, columnar_t, dense_t, invalid_id_t>, TypeCombo<columnar_t, columnar_t, sparse_t, invalid_id_t>,
+    TypeCombo<columnar_t, row_t, dense_t, invalid_id_t>, TypeCombo<columnar_t, row_t, sparse_t, invalid_id_t>,
+    TypeCombo<row_t, columnar_t, dense_t, invalid_id_t>, TypeCombo<row_t, columnar_t, sparse_t, invalid_id_t>) {
```

Comment:
it looks a bit heavy at first sight but looking at the rest of the tests, this is a significant improvement :+1: